utopia-remix/app/services/fgaService.server.ts

```diff
@@ -0,0 +1,104 @@
+import { CredentialsMethod, OpenFgaClient } from '@openfga/sdk'
+import { ServerEnvironment } from '../env.server'
+import { AccessLevel } from '../types'
+
+export const fgaClient = new OpenFgaClient({
+  apiScheme: 'https',
+  apiHost: 'api.us1.fga.dev',
+  storeId: ServerEnvironment.FGA_STORE_ID,
+  // authorizationModelId: 'YOUR_MODEL_ID', // Optionally, you can specify a model id to target, which can improve latency
+  credentials: {
+    method: CredentialsMethod.ClientCredentials,
+    config: {
+      apiTokenIssuer: 'fga.us.auth0.com',
+      apiAudience: 'https://api.us1.fga.dev/',
+      clientId: ServerEnvironment.FGA_CLIENT_ID,
+      clientSecret: ServerEnvironment.FGA_SECRET,
+    },
+  },
+})
+
+export async function updateAccessLevel(projectId: string, accessLevel: AccessLevel) {
+  switch (accessLevel) {
+    case AccessLevel.PUBLIC:
+      await fgaClient.write({
+        writes: [
+          {
+            user: 'user:*',
+            relation: 'viewer',
+            object: `project:${projectId}`,
+          },
+        ],
+      })
+      break
+    case AccessLevel.PRIVATE:
+      await fgaClient.write({
+        deletes: [
+          {
+            user: 'user:*',
+            relation: 'viewer',
+            object: `project:${projectId}`,
+          },
+        ],
+      })
+      break
+  }
+}
+
+const userProjectPermission = [
+  'can_view',
+  'can_fork',
+  'can_play',
+  'can_edit',
+  'can_comment',
+  'can_show_presence',
+  'can_see_live_changes',
+  'can_request_access',
+] as const
+
+type UserProjectPermission = (typeof userProjectPermission)[number]
+
+async function checkUserProjectPermission(
+  projectId: string,
+  userId: string,
+  permission: UserProjectPermission,
+) {
```

Comment:
could you add explicit return types here, in the `can<Action>Project` functions please? I'm trying to be more diligent in that regard myself, I think it'd be beneficial to have it consistent across the board